src/main/java/service/activity/SupplyHandler.java

```diff
@@ -0,0 +1,13 @@
+package service.activity;
+
+import java.util.Map;
+import model.FruitRecord;
+
+public class SupplyHandler implements ActivityHandler {
+
+    @Override
+    public void use(FruitRecord fruitRecord, Map<String, Integer> report) {
+        report.put(fruitRecord.getFruitName(),
+                CheckNullBalance.get(fruitRecord.getFruitName(), report) + fruitRecord.getAmount());
```

Comment:
you can check for null here as well
or at least make method non static
static methods are bad practice in most cases